File: android/Buyo/app/src/main/java/com/cmpe352group4/buyo/vo/Product.kt
package com.cmpe352group4.buyo.vo

import com.cmpe352group4.buyo.vo.Comment
import com.cmpe352group4.buyo.vo.FilterCriterias
import com.cmpe352group4.buyo.vo.ProductInfo
import com.cmpe352group4.buyo.vo.Vendor
import java.io.Serializable

data class Product(
    var id: String,
    var category: List<String>,
    var name: String,
    var imageUrl: String,
    var rating: Double,
    var price: Double,
    var originalPrice: Double,
    var brand: String,
    var productInfos: List<ProductInfo>,
    var vendor: Vendor,
    var description: String,
    var materials: List<String>?,
    var comments : List<Comment>?,
    var filterCriterias: List<FilterCriterias>?
): Serializable

File: backend/product_service/models/product.js
const mongoose = require("mongoose");

/**
 * Schema of a Vendor object in the database, vendor documents resides
 * in 'vendors' collection.
 */
let schema =  new mongoose.Schema({
  name: String,
  imageUrl: String,
  category: [String],
  rating: Number,
  price: Number,
  originalPrice: Number,
  productInfos: String,
  description: String,
  brand: String,
  vendorId: mongoose.Schema.Types.ObjectId,
});
schema.index({ name: 'text', category: 'text', description: 'text', brand: 'text' });

module.exports.Product = mongoose.model(
  "Product",
  schema,
  "products"
);
